File: src/leetcode/python/LongestPalindrome_409.py
class LongestPalindrome_409(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        chars = {}
        for c in s:
            if c not in chars:
                chars[c] = 1
            else:
                chars[c] += 1
        l, fg = 0, False
        for ch in chars:
            f = chars.get(ch)
            if f % 2 == 0:
                l += f
            else:
                l += (f - 1)
                fg = True
        return l if not fg else l + 1

File: src/leetcode/java/LongestPalindrome_409.java
package leetcode.java;

public class LongestPalindrome_409 {
  public int longestPalindrome(String s) {
    int[] chs = new int[52];
    for (char c : s.toCharArray()) {
      if (Character.isUpperCase(c)) chs[c - 'A' + 26]++;
      else chs[c - 'a']++;
    }
    boolean odd = false;
    int length = 0;
    for (int c : chs) {
      if (c % 2 == 0) length += c;
      else {
        length += (c - 1);
        odd = true;
      }
    }
    return odd ? length + 1 : length;
  }
}
